### PR TITLE
[coverage] Test more SuperSockets

### PR DESCRIPTION
**Pull request by guedou**
This PR adds test for `StreamSocket` and `L3RawSocket`.

### CODE REVIEW

**Comment by gpotter2** [comment]
@guedou If you are willing to test `L2ListenTcpdump`, the current windows implementation is broken.
On windows, `windump` has a very poor/broken auto-selection of the interface, so you'll need to pass it manually. Also, you need to pass the `pcap_name` arg as so:
```
diff --git a/scapy/supersocket.py b/scapy/supersocket.py
index 8acfaa65..8b38e4d6 100644
--- a/scapy/supersocket.py
+++ b/scapy/supersocket.py
@@ -192,7 +192,15 @@ class L2ListenTcpdump(SuperSocket):
         self.outs = None
         args = ['-w', '-', '-s', '65535']
         if iface is not None:
-            args.extend(['-i', iface])
+            if WINDOWS:
+                try:
+                    args.extend(['-i', iface.pcap_name])
+                except AttributeError:
+                    args.extend(['-i', iface])
+            else:
+                args.extend(['-i', iface])
+        elif WINDOWS:
+            args.extend(['-i', conf.iface.pcap_name])
         if not promisc:
             args.append('-p')
         if not nofilter:
```

**Comment by guedou** [comment]
@p-l- ready to be reviewed. Thanks !

File: scapy/sendrecv.py [hunk 1]
```diff
@@ -94,8 +94,9 @@ def _get_pkt():
         def _get_pkt():
             if bpf_select([pks]):
                 return pks.recv()
-    elif conf.use_pcap or (not isinstance(pks, StreamSocket)
-                           and (DARWIN or FREEBSD or OPENBSD)):
+    elif (conf.use_pcap and not isinstance(pks, StreamSocket) and
```
Comment (p-l-) [inline scapy/sendrecv.py]: `not isinstance(pks, StreamSocket) and not isinstance(pks, L3RawSocket)` can be written as `not isinstance(pks, (StreamSocket, L3RawSocket))`.

File: test/regression.uts [hunk 2]
```diff
@@ -7538,6 +7538,29 @@ assert(p[Ether:2].type == 0x800)
...
++ Tests of StreamSocket
+
+= Test with DNS over TCP
+~ netaccess
+
+import socket
+sck = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+sck.connect(("8.8.8.8", 53))
+
+class DNSTCP(Packet):
```
Comment (p-l-) [inline test/regression.uts]: We should be able to use the regular `DNS` layer for that, I'll fix that later.

Comment (guedou) [reply test/regression.uts]: I did not find an easy way to achieve the same with `DNS()` without a TCP underlayer.

**Pull request merged**